### PR TITLE
Removed omit id on FirstOrCreate

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
Omitting the ID on gorm FirstOrCreate results in not fetching the records correctly if they exist.
This causes multiple records being saved for the same project domain attribute.

Also added the missing update existing record test.

Added index due to which multiple records were being saved. This was an issue after gorm v2 upgrade.
```
flyteadmin=# \d resources;
                                        Table "public.resources"
    Column     |           Type           | Collation | Nullable |                Default                
---------------+--------------------------+-----------+----------+---------------------------------------
 id            | bigint                   |           | not null | nextval('resources_id_seq'::regclass)
 created_at    | timestamp with time zone |           |          | 
 updated_at    | timestamp with time zone |           |          | 
 deleted_at    | timestamp with time zone |           |          | 
 project       | text                     |           |          | 
 domain        | text                     |           |          | 
 workflow      | text                     |           |          | 
 launch_plan   | text                     |           |          | 
 resource_type | text                     |           |          | 
 priority      | integer                  |           |          | 
 attributes    | bytea                    |           |          | 
Indexes:
    "resources_pkey" PRIMARY KEY, btree (id)
    "resource_idx" UNIQUE, btree (project, domain, workflow, launch_plan, resource_type)
```

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/2210

## Follow-up issue
_NA_

### CODE REVIEW

**Review by katrogan — APPROVED** [review]
nice find! can you add an integration test here too? https://github.com/flyteorg/flyteadmin/tree/master/tests

File: pkg/repositories/models/resource.go [hunk 1]
```diff
@@ -18,11 +18,11 @@ type Resource struct {
...
-	Project      string     `gorm:"unique_index:resource_idx" valid:"length(0|255)"`
-	Domain       string     `gorm:"unique_index:resource_idx" valid:"length(0|255)"`
-	Workflow     string     `gorm:"unique_index:resource_idx" valid:"length(0|255)"`
-	LaunchPlan   string     `gorm:"unique_index:resource_idx" valid:"length(0|255)"`
-	ResourceType string     `gorm:"unique_index:resource_idx" valid:"length(0|255)"`
+	Project      string     `gorm:"uniqueIndex:resource_idx" valid:"length(0|255)"`
+	Domain       string     `gorm:"uniqueIndex:resource_idx" valid:"length(0|255)"`
+	Workflow     string     `gorm:"uniqueIndex:resource_idx" valid:"length(0|255)"`
+	LaunchPlan   string     `gorm:"uniqueIndex:resource_idx" valid:"length(0|255)"`
+	ResourceType string     `gorm:"uniqueIndex:resource_idx" valid:"length(0|255)"`
```
Comment (EngHabu) [inline pkg/repositories/models/resource.go:25]: Are these equivalent? or did we never have uniqe_index correctly? or is it deprecated...?

Comment (pmahindrakar-oss) [reply pkg/repositories/models/resource.go:25]: @EngHabu unique_index worked in v1 but in v2 they changed it uniqueIndex

https://gorm.io/docs/indexes.html#uniqueIndex
https://v1.gorm.io/docs/models.html

This is one of the other breaking changes along with auto_increment which we faced during v2 migration

https://gorm.io/docs/v2_release_note.html#Breaking-Changes

Should have checked this out during the migration :)

**Comment by pmahindrakar-oss** [comment]
Done @katrogan
